doc/contribute/project_roles.rst

```diff
@@ -0,0 +1,281 @@
+* All submitted pull-requests should have an assignee
+* We Introduce a maintainers' hierarchy to address the above point
+
+  * Parent-area maintainer should be acting as default substitute/fallback assignee for un-maintained sub-areas
+  * Area maintainer gets precedence over parent-area maintainer
+
+* Pull-requests may be re-assigned if this is needed or more appropriate
+
+  * Re-assigned by original assignee (see “Assignee” slide)
+
+* Updates to the maintainers’ file should be in a standalone PRs
+* The Maintainer’s file itself shall have a maintainer
+* Architectures, core components, sub-systems, samples, tests
+
+  * Each area shall have an explicit maintainer
+
+* Boards (incl relevant samples, tests), SoCs (incl DTS)
+  * May have a maintainer, shall have a higher-level platform maintainer
+* Drivers
+
+  * Shall have a driver-area (and API) maintainer
+  * Could have individual driver implementation maintainers but preferably collaborator/contributors
+  * In the above case, platform-specific PRs may be re-assigned to respective collaborator/contributor of driver implementation
+
+
+Merge Criteria
+##############
+
+* All continuous integration checks have passed
+
+  * Codeowners
+  * Device Tree
+  * Documentation
+  * Gitlint
+  * Identity/Emails
+  * Kconfig
+  * License
+  * Checkpatch (Coding Style)
+  * Pylint
+  * Sanitycheck + Other Unit tests
+  * Simulated Bluetooth Tests
+
+* Planned
+
+  * Footprint
+  * Code coverage
+  * Coding Guidelines
+  * Static Analysis (Coverity)
+  * Documentation coverage (APIs)
+
+* PR template with checklist
+
+* Minimal of 2 approvals
+
+  * A collaborator from the same subsystem.
+  * Alternately another maintainer of another subsystem
+  * Approval by the assignee
+
+* A minimum review period of 2 days, 4 hours for trivial changes.
+  Hotfixes can be merged at any time after CI passes.
+* All required checks are passing
+
+Escalation Process
+##################
+
+* Contributors may object to change requests or decisions made by Maintainers.
+* Process
+
+  * Resolve in the PR among assignee, maintainers and reviewer
+
+    * Assignee to act as moderator if applicable
+
+  * Optionally resolve in the dev review meeting with more maintainers and project stakeholders
+
+    * Assignee to be present when the (escalated) issue is discussed
+
+  * TSC: Assignees can escalate to the TSC voting members and get a binding resolution in the TSC.
+  * Assignee to ensure the resolution of the escalation is reflected in the PR review.
+
+
+Release Manager
+###############
+
+A *Maintainer* responsible for driving a specific release to
+completion following the milestones and the roadmap of the
+project for this specific release.
+
+* TSC has to approve a release manager.
+
+A Release Manager joins the release engineering team and has
+the rights and responsibilities of that team in addition to
+the following:
+
+* Right to manage and coordinate all code merges after the
+  code freeze milestone (M3)
```

Comment:
Would it make sense linking to https://github.com/zephyrproject-rtos/zephyr/wiki/Program-Management ?

Reply:
I think this can be appended, yes.
I'll add this on the amended proposal.